### DISCUSSION TITLE
Help) DECK_INFO: Warning! No driver found for deck.

### DISCUSSION THREAD

**Discussion by jeongheonn** [bitcraze/discussions#1474] [Q&A]
Hello Bitcraze,

We are using the Crazyflie 2.1, Flow Deck v2, Crazyradio, and Lighthouse Positioning Deck. When using the official Crazyflie firmware, we didn't encounter any issues with deck detection, and it was confirmed in the console tab of the cfclient.

However, we want to use an external firmware called tinympc-crazyflie-firmware. When we make and make cload the example controller_tinympc_state_soc from this firmware, we receive an error message indicating that the driver cannot be found, as shown in the attached photo.

We have limited knowledge of firmware, so please bear with us. Is there something additional we need to include when making the firmware to support the relevant driver?

Here is the link to the firmware we're using.

https://github.com/RoboticExplorationLab/tinympc-crazyflie-firmware

Thank you.

```

<crazyflie-firmware, with flow deck> -----> No problem detected 
SYS: Crazyflie 2.1 is up and running!
SYS: Build 0:e527895e8c6e (2023.11) CLEAN
SYS: I am 0x203235324133501900420033 and I have 1024KB of flash!
CFGBLK: v1, verification [OK]
DECK_CORE: 1 deck(s) found
DECK_CORE: Calling INIT on driver bcFlow2 for deck 0
ZR2: Z-down sensor [OK]
PMW: Motion chip id: 0x49:0xB6
IMU: BMI088: Using I2C interface.
IMU: BMI088 Gyro connection [OK].
IMU: BMI088 Accel connection [OK]
IMU: BMP388 I2C connection [OK]
ESTIMATOR: Using Kalman (2) estimator
CONTROLLER: Using PID (1) controller
MTR-DRV: Using brushed motor driver
SYS: About to run tests in system.c.
SYS: NRF51 version: 2024.2 (CF21)
EEPROM: I2C connection [OK].
STORAGE: Storage check [OK].
IMU: BMI088 gyro self-test [OK]
DECK_CORE: Deck 0 test [OK].
SYS: Self test passed!
STAB: Wait for sensor calibration...
SYS: Free heap: 17608 bytes
STAB: Starting stabilizer loop
ESTKALMAN: State out of bounds, resetting
SUP: Can not fly
SUP: Ready to fly

```

```


<tinympc-crazyflie-firmware with flow deck> ------> No driver found for deck 
SYS: Crazyflie 2.1 is up and running!
SYS: Build 0:5af0a2f189fe (NA) CLEAN
SYS: I am 0x203235324133501900420033 and I have 1024KB of flash!
CFGBLK: v1, verification [OK]
DECK_INFO: Warning! No driver found for deck.
DECK_CORE: 1 deck(s) found
IMU: BMI088: Using I2C interface.
IMU: BMI088 Gyro connection [OK].
IMU: BMI088 Accel connection [OK]
IMU: BMP388 I2C connection [OK]
ESTIMATOR: Using Complementary (1) estimator
CONTROLLER: Using PID (1) controller
MTR-DRV: Using brushed motor driver
SYS: About to run tests in system.c.
SYS: NRF51 version: 2024.2 (CF21)
EEPROM: I2C connection [OK].
STORAGE: Storage check [OK].
IMU: BMI088 gyro self-test [OK]
SYS: Self test passed!
STAB: Wait for sensor calibration...
SYS: Free heap: 20504 bytes
MPC: Waiting for activation ...
STAB: Ready to fly.
STAB: WARNING: stabilizer loop rate is off (501)

```

![Screenshot from 2024-07-18 08-59-33](https://github.com/user-attachments/assets/6346b5c5-d13a-4b63-bb13-a2e7c8c37425)
(Screenshot when we attached flow deck and Ai deck)

**Comment by evoggy** [comment]
Hi! Happy to hear that you're trying this out! If the stock firmware works I would recommend posting about this issue in [tinympc](https://github.com/RoboticExplorationLab/tinympc-crazyflie-firmware) repository instead since this seems to be where the issue is.

**Comment by whY-note** [comment]
Hello, jeongheonn:
 
I had the exact same problem as you.
Have you solved this problem?
If so, could you help me?

Your prompt reply will be highly appreciated!

Thank you!

**Reply by jeongheonn** [reply]
please check email! Please let me know if the issue has been resolved successfully.

**Comment by 1999leedongmin** [comment]
Hi jeongheonn,

I’m a master’s student at Inha University working with Crazyflie 2.1 and the Lighthouse positioning system. I’m trying to use TinyMPC for trajectory tracking, but I’m running into some issues.

Right now, I’ve set up the Lighthouse system and loaded the TinyMPC firmware to test hovering. However, the Lighthouse connection seems unstable, and I suspect it might be affecting the positioning.

Could you share what firmware version you used and how you set up the Lighthouse system? It would really help me troubleshoot!

Looking forward to your advice.

Thanks!

**Reply by jeongheonn** [reply]
메일 확인 부탁드립니다!
